udata/core/spatial/models.py

```diff
@@ -96,19 +147,24 @@ def postal_string(self):
         """Return a list of postal codes separated by commas."""
         return ', '.join(self.keys.get('postal', []))
 
-    @cached_property
-    def town_repr(self):
-        """Representation of a town with optional county."""
-        if self.county:
-            return '{name} <small>({county_name})</small>'.format(
-                name=self.name, county_name=self.county.name)
-        return self.name
+    @property
+    def parent(self):
+        if self.parent_level:
+            for parent in self.parents:
+                if parent.startswith(self.parent_level):
```

Comment:
No, parent can be `fr/region/53` and `parent_level` will be `fr/region` for instance.